ui/jobs/components/ast.ts

```diff
@@ -62,34 +62,148 @@ export class ASTComponent extends BaseComponent {
+    if (sealCount === 3)
+      this.sealBox.parentNode.classList.add('ready');
+    else
+      this.sealBox.parentNode.classList.remove('ready');
+  }
+
+  override onUseAbility(id: string): void {
+    switch (id) {
+      case kAbility.Combust2:
+      case kAbility.Combust3:
+        this.combustBox.duration = 30;
+        break;
+      case kAbility.Combust:
+        this.combustBox.duration = 18;
+        break;
+      case kAbility.Draw:
+        this.drawBox.duration = 30;
+        break;
+      case kAbility.LucidDreaming:
+        this.lucidBox.duration = 60;
+        break;
+    }
+  }
+  override onStatChange({ gcdSpell }: { gcdSpell: number }): void {
+    this.combustBox.valuescale = gcdSpell;
+    this.combustBox.threshold = gcdSpell + 1;
+    this.drawBox.valuescale = gcdSpell;
+    this.drawBox.threshold = gcdSpell + 1;
+    this.lucidBox.valuescale = gcdSpell;
+    this.lucidBox.threshold = gcdSpell + 1;
+  }
+
+  override reset(): void {
+    this.combustBox.duration = 0;
+    this.drawBox.duration = 0;
+    this.lucidBox.duration = 0;
+  }
+}
+
+export class ASTComponent extends BaseComponent {
+  combustBox: TimerBox;
+  drawBox: TimerBox;
+  lucidBox: TimerBox;
+  cardBox: ResourceBox;
+  minorBox: ResourceBox;
+  signs: HTMLElement[] = [];
+
+  constructor(o: ComponentInterface) {
+    super(o);
+
+    this.combustBox = this.bars.addProcBox({
+      id: 'ast-procs-combust',
+      fgColor: 'ast-color-combust',
+      notifyWhenExpired: true,
+    });
+
+    this.drawBox = this.bars.addProcBox({
+      id: 'ast-procs-draw',
+      fgColor: 'ast-color-draw',
+    });
+
+    this.lucidBox = this.bars.addProcBox({
+      id: 'ast-procs-luciddreaming',
+      fgColor: 'ast-color-lucid',
+    });
+
+    this.cardBox = this.bars.addResourceBox({
+      classList: ['ast-color-card'],
+    });
+
+    this.minorBox = this.bars.addResourceBox({
+      classList: ['ast-color-card'],
+    });
+
+    // Sign
+    const stacksContainer = document.createElement('div');
+    stacksContainer.id = 'ast-stacks';
+    stacksContainer.classList.add('stacks');
+    this.bars.addJobBarContainer().appendChild(stacksContainer);
+    const signContainer = document.createElement('div');
+    signContainer.id = 'ast-stacks-sign';
+    stacksContainer.appendChild(signContainer);
+
+    for (let i = 0; i < 3; ++i) {
+      const d = document.createElement('div');
+      signContainer.appendChild(d);
+      this.signs.push(d);
+    }
+
+    this.reset();
+  }
+
+  override onJobDetailUpdate(jobDetail: JobDetail['AST']): void {
+    // Minor Arcana, ↑ = lord, + = lady
+    const minor = jobDetail.crownCard;
+    this.minorBox.parentNode.classList.toggle('lord', minor === 'Lord');
+    this.minorBox.parentNode.classList.toggle('lady', minor === 'Lady');
+    if (minor === 'Lord')
+      this.minorBox.innerText = '↑';
+    else if (minor === 'Lady')
+      this.minorBox.innerText = '＋';
+    else
+      this.minorBox.innerText = '';
```

Comment:
This might be simplified as this better.

```suggestion
    this.minorBox.innerText = minorMap[minor] ?? '';
```

and add this to the top:
```ts
const minorMap = {
  'Lord': '↑',
  'Lady': '＋',
}
```